### PR TITLE
Basic Dashboard View

### PR DESCRIPTION
**Pull request by 3ygun**
Started Issue #12
Added a Dashboard view and updated some basic UI on the site

Current issues
- [x] Delete doesn't redirect properly by to the dashboard
- [x] Basic dashboard

### CODE REVIEW

File: app/controllers/home_controller.rb [hunk 1]
```diff
@@ -1,7 +1,38 @@
...
     redirect_to time_entries_url if user_signed_in?
   end
+
+  def dashboard
+    ## Main objects
+    @admin = current_user.admin?
+    @date = params[:date] ? Date.parse(params[:date]) : Date.today
+    @tasks = Task.includes(:tags)
+    @time_entries = TimeEntry.all
+    @tags = @admin ? Tag.all : Tag.where(user: current_user)
```
Comment (J3RN) [inline app/controllers/home_controller.rb]: While this ternary syntax is more concise, I think I prefer the syntax you used for `@time_entries` and `@tasks` better. To consolidate, we could change this section to
```ruby
@tasks = Task.includes(:tags)
@time_entries = TimeEntry.all
@tags = Tag.all
```

And then below have:
```ruby
# Limit to just ours if we're not an admin
unless @admin
  @tasks = @tasks.where(user: current_user)
  @time_entries = @time_entries.where(user: current_user)
  @tags.where(user: current_user)
end

File: app/controllers/tasks_controller.rb [hunk 2]
```diff
@@ -4,15 +4,6 @@ class TasksController < ApplicationController
   before_action :set_users, only: [:new, :edit]
   before_action ->{ ensure_ownership(@task) }, only: [:edit, :update, :destroy]
 
-  def index
```
Comment (J3RN) [inline app/controllers/tasks_controller.rb]: While having `index` views for things doesn't make sense in the Unified Dashboard™ view of things, we'll still need to have this index view for the eventual API. So, my suggestion would be to just leave this here and just not link to this from anywhere. That, or we could par this method down to
```ruby
redirect_to :dashboard
```

**Comment by 3ygun** [comment]
Made @J3RN 's suggested changes and fixed the deletion redirection/edit bug :+1: 

Awaiting review upon which can/should be merged.

**Comment by J3RN** [comment]
Hey David! I'm glad to see you finished this! I'm going to be super busy over the next few days, but I should be able to review and merge this on Friday. In the meantime - Since it's finished, you can remove the Work in Progress ("WIP") label. Also, you should update the description to not include the Todo items you've finished.

PROTIP: Markdown supports checkboxes for doing todo-lists easily, like so:
```
- [ ] Todo item
- [x] Done item
```

And it renders to:

- [ ] Todo item
- [x] Done item

File: app/controllers/tasks_controller.rb [hunk 3]
```diff
@@ -34,12 +28,17 @@ def create
...
+    redirect_to dashboard_path
   end
 
   def destroy
     @task.destroy
-    redirect_to tasks_path
+    redirect_to dashboard_path
+  end
+
+  def delete
```
Comment (J3RN) [inline app/controllers/tasks_controller.rb]: The REST `DELETE` verb should actually call the `destroy` method above, so you shouldn't need to have this method.

File: app/views/tasks/_form.html.erb [hunk 4]
```diff
@@ -43,4 +43,12 @@
   <div class="actions">
     <%= f.submit(nil, class: "btn btn-primary") %>
   </div>
+  <div>
+    <% if @task.id == nil %>
+      <%= link_to 'Back', dashboard_path, data: { confirm: 'Are you sure?' }, class: 'btn btn-danger' %>
```
Comment (J3RN) [inline app/views/tasks/_form.html.erb]: We need to check that I'm sure that I want to go back? I can understand the sentiment of "This stuff won't be saved if you leave," so I would suggest changing this to something that sounds more like that.

File: test/controllers/home_controller_test.rb [hunk 5]
```diff
@@ -1,8 +1,24 @@
...
+    @user = users(:one)
+    sign_in @user
+
+    @task = tasks(:one)
+  end
+
   test "should get index" do
     get :index
-    assert_response :success
+    # TODO - Fix?
```
Comment (J3RN) [inline test/controllers/home_controller_test.rb]: Well, if the user is signed in, per your `setup` block, they won't land on the `index` page. They'll be redirected somewhere. Maybe you could just sign them in as part of the dashboard test?

File: test/controllers/home_controller_test.rb [hunk 6]
```diff
@@ -1,8 +1,24 @@
...
+  end
+
+  # test "should not see others' tasks" do
+  '''
+    get :dashboard
+
+    assert_equal(@user.tasks.active.count, assigns(:active).count)
+    assert_equal(@user.tasks.archived.count, assigns(:archived).count)
```
Comment (J3RN) [inline test/controllers/home_controller_test.rb]: Tests aren't very helpful when they're commented out!

*P.S. I haven't seen block comments in Ruby in so long, I had to test what these `'''` things did.*

**Comment by J3RN** [comment]
Alright, so here's my commentary:

Essentially, this PR just moves the Tasks controller to the Home controller, which doesn't really add any value. I'd like to get some more stuff in here before we merge it.

Also, the Tasks controller is still accessible (`localhost:3000/tasks`), but it throws an ugly error since you moved it's view.
screen shot 2016-03-21 at 08 59 46
 If you want to completely remove the `index` view of the Tasks controller, change it's routes from `resources :tasks` to `resources :tasks, except: :index`, and delete the `index` method in the Tasks controller.

File: app/views/tasks/_form.html.erb (hunk 4 above)
Comment (3ygun) [inline app/views/tasks/_form.html.erb]: You completed this in #34

File: app/controllers/tasks_controller.rb (hunk 3 above)
Comment (3ygun) [inline app/controllers/tasks_controller.rb]: Done

**Comment by 3ygun** [comment]
Completed all of @J3RN's comments. Just need to fix the test soon :tm:

**Comment by J3RN** [comment]
Since this has gone dormant, I'm going to close it for now. We can re-open it later if necessary.

**Pull request closed without merging**